backend/drm/renderer.c

```diff
@@ -212,8 +204,15 @@ static struct wlr_drm_format *create_linear_format(uint32_t format) {
 bool drm_plane_init_surface(struct wlr_drm_plane *plane,
 		struct wlr_drm_backend *drm, int32_t width, uint32_t height,
 		uint32_t format, bool with_modifiers) {
+	const struct wlr_pixel_format_info *info =
+		drm_get_pixel_format_info(format);
+	if (!info) {
```

Comment:
If we don't have the format in our table, it doesn't mean we can't use it in DRM. It just means we can't fallback to the same format without the alpha channel, because we don't know what it is.

A better solution would be to only fallback to the opaque substitute if `info != NULL`.